docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,304 @@
+
+**Quick test checklist:**
+- [ ] Run `./scripts/lint.sh` for code quality checks
+- [ ] Run `pytest --cov=src` for unit tests with coverage
+- [ ] Build and test your container image
+- [ ] Verify component works with sample data
+
+### 6. Commit Your Changes
+
+```bash
+# Stage your changes
+git add .
+
+# Check what you're committing
+git status
+git diff --cached
+
+# Commit with descriptive message following Conventional Commits
+git commit -m "feat: add data preprocessing component
+
+- Implements StandardScaler and MinMaxScaler
+- Adds comprehensive unit and integration tests
+- Includes usage examples and documentation
+- Resolves #123"
+
+# For bug fixes:
+git commit -m "fix: resolve memory leak in data loader
+
+- Fix buffer overflow in large dataset processing
+- Add memory usage tests
+- Update documentation with memory requirements
+- Fixes #456"
+```
+
+### 7. Push and Create Pull Request
+
+```bash
+# Push to your fork
+git push origin component/my_component
+
+# If this is your first push for this branch:
+git push --set-upstream origin component/my_component
+```
+
+**Create PR on GitHub:**
+1. Navigate to your fork on GitHub
+2. Click "Compare & pull request" button
+3. Fill out the PR template:
+   - **Title**: Clear, descriptive title
+   - **Description**: What does this PR do?
+   - **Testing**: How was this tested?
+   - **Checklist**: Complete the provided checklist
+   - **Related Issues**: Link to relevant issues
+
+
+## Testing Requirements
+
+*Minimum testing standards all components must meet.*
+
+- **Unit tests**: Test individual functions
+- **Integration tests**: Test end-to-end functionality  
+- **Coverage**: Minimum 80%
+- **Error handling**: Test failure scenarios
+
+See [TESTING.md](TESTING.md) for detailed testing guidelines.
+
+## Review Process
+
+*How contributions are reviewed and what criteria they must meet.*
+
+All PRs must pass:
+- Automated checks (linting, tests, builds)
+- Code review by maintainers
+- Documentation review
+
+### Review Criteria
+
+- Component works as described
+- Code is clean and well-documented
+- Tests provide good coverage
+- Follows repository standards
+
+## Before Submitting
+
+*Final checklist to ensure your contribution is ready for review.*
+
+- [ ] Component builds successfully
+- [ ] All tests pass (≥80% coverage)
+- [ ] All required files present
+- [ ] Documentation complete
+- [ ] Follows naming conventions
+- [ ] No security vulnerabilities
+
+## Getting Help
+
+*Resources and channels for getting support during development.*
+
+- **Setup**: [ONBOARDING.md](ONBOARDING.md)
+- **Testing**: [TESTING.md](TESTING.md)
+- **Governance**: [GOVERNANCE.md](GOVERNANCE.md)
+- **Best Practices**: [BESTPRACTICES.md](BESTPRACTICES.md) *(coming soon)*
+- **Agents**: [AGENTS.md](AGENTS.md) *(coming soon)*
+- **Community**: [#kubeflow-pipelines Slack](https://kubeflow.slack.com/channels/kubeflow-pipelines)
```

Comment:
This is the old KFP Slack. If we point to the website as a canonical link, we prevent it from being outdated in the future.

```suggestion
- **Community**: `#kubeflow-pipelines` channel on the [CNCF Slack](https://www.kubeflow.org/docs/about/community/#kubeflow-slack-channels)
```